cmake/modules/BuildQAT.cmake

```diff
@@ -0,0 +1,48 @@
+function(build_qat)
+  set(QAT_BINARY_DIR ${CMAKE_BINARY_DIR}/src/qatlib)
+  set(QAT_INSTALL_DIR ${QAT_BINARY_DIR}/install)
+  set(QAT_INCLUDE_DIR ${QAT_INSTALL_DIR}/include)
+  set(QAT_LIBRARY_DIR ${QAT_INSTALL_DIR}/lib)
+  set(QAT_LIBRARY ${QAT_LIBRARY_DIR}/libqat.a)
+  set(QAT_USDM_LIBRARY ${QAT_LIBRARY_DIR}/libusdm.a)
+
+  # this include directory won't exist until the install step, but the
+  # imported targets need it early for INTERFACE_INCLUDE_DIRECTORIES
+  file(MAKE_DIRECTORY "${QAT_INCLUDE_DIR}")
+
+  set(configure_cmd env CC=${CMAKE_C_COMPILER} ./configure --prefix=${QAT_INSTALL_DIR})
+  # disable systemd or 'make install' tries to write /usr/lib/systemd/system/qat.service
+  list(APPEND configure_cmd --disable-systemd)
+  # build a static library with -fPIC that we can link into crypto/compressor plugins
+  list(APPEND configure_cmd --with-pic --enable-static --disable-shared)
+
```

Comment:
@cbodley suggesting
```suggestion
# disable the building of samples
list(APPEND configure_cmd --enable-samples=no)

```
tested to work locally
```
less ./qatlib_ext-prefix/src/qatlib_ext-stamp/qatlib_ext-configure.log
...
checking force disable fast crc in assembler... no
checking whether to build samples... no
                                     ^^
checking that generated files are newer than configure... done
...
```

Reply:
what was the problem with samples? the ci builds in https://shaman.ceph.com/builds/ceph/wip-cmake-qat/bf90dbbbfb6ec1b998369749506e590d3d9dbac6/ seem ok there